postmark.go

```diff
@@ -43,35 +53,64 @@ func New(serverToken, accountToken string) Postmark {
 	}
 }
 
-func (p *postmark) exec(ctx context.Context, r *http.Request) error {
+func (p *postmark) Templates() Templates {
+	return &templates{pm: p}
+}
+
+func (p *postmark) Emails() Emails {
+	return &emails{pm: p}
+}
+
+func (p *postmark) Exec(ctx context.Context, req *Request) (*http.Response, error) {
+	var payload io.Reader
+	if req.Payload != nil {
+		data, err := json.Marshal(req.Payload)
+		if err != nil {
+			return nil, err
+		}
+
+		payload = bytes.NewReader(data)
+	}
+
+	r, err := http.NewRequest(req.Method, path.Join(rootEndpoint, req.Path), payload)
```

Comment:
This will put a `/` between the path and the host, so there's no need for the `/` leading all of the path constructions. https://play.golang.org/p/b1db5j7j6E